mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+import mlflow
+from mlflow.models.evaluation.base import (
+    ModelEvaluator,
+    EvaluationMetrics,
+    EvaluationArtifact,
+)
+from mlflow.utils.file_utils import TempDir
+from mlflow.tracking.artifact_utils import get_artifact_uri
+from sklearn import metrics as sk_metrics
+import matplotlib.pyplot as pyplot
+import scikitplot
+import shap
+import math
+
+shap.initjs()
+
+"""
+[P0] Accuracy: Calculates how often predictions equal labels.
+[P0] BinaryCrossentropy: Computes the crossentropy metric between the labels and predictions.
+[P0] Hinge: Computes the hinge metric between y_true and y_pred.
+[P0] Sum: Computes the (weighted) sum of the given values.
+[P0] Mean: Computes the (weighted) mean of the given values.
+[P0] ExampleCount: Computes the total number of evaluation examples.
+[P0] MeanAbsoluteError: Computes the mean absolute error between the labels and predictions.
+[P0] MeanSquaredError: Computes the mean squared error between y_true and y_pred.
+[P0] RootMeanSquaredError: Computes root mean squared error metric between y_true and y_pred.
+
+[P0] TrueNegatives: Calculates the number of true negatives.
+[P0] TruePositives: Calculates the number of true positives.
+[P0] FalseNegatives: Calculates the number of false negatives.
+[P0] FalsePositives: Calculates the number of false positives.
+https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix
+
+[P0] Precision: Computes the precision of the predictions with respect to the labels.
+[P0] Recall: Computes the recall of the predictions with respect to the labels.
+[P0] AUC: Approximates the AUC (Area under the curve) of the ROC or PR curves.
+[P0] F1 Score: 2*precision*recall / (precision+recall)
+
+[P0] BinaryClassConfusionMatrix
+
+Plots
+[P0] Confusion matrix
+[P0] Interactive ROC curve with metrics (TP/TN/FP/FN/Acc/F1/AUC), binary classification
+[P0] Lift chart
+
+Global explainability
+[P0] Model built-in feature importance (supported models)
+[P0] SHAP explainers
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+
+    @classmethod
+    def save_content_to_file(cls, content, output_artifact_path):
+        assert isinstance(content, Image)
+        content.save(output_artifact_path)
+
+    @classmethod
+    def load_content_from_file(cls, local_artifact_path):
+        return open_image(local_artifact_path)
+
+
+class DefaultEvaluator(ModelEvaluator):
+
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_image_artifact(self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name):
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
```

Comment:
may want to add image size adjustments for anything related to matplotlib. Most visualizations in these packages do auto-scaling of content to make everything fit and if the resolution and image size isn't adjusted based on the content payload (i.e., the number of features) the plot will be illegible (font size 1, for instance)

Reply:
might want to also think about dpi resolution for images - the default (300, I think?) is pretty low and makes rendering from resizing look really bad.

Reply:
@WeichenXu123 Can we look into this?

Reply:
For reference, here's some fairly complex logic we used for feature importance plot sizing in XGBoost: https://github.com/mlflow/mlflow/blob/c630d6d2f0819bd873106faebe7f9a26554b2aa5/mlflow/xgboost/__init__.py#L447-L521. We should at least try out the current `pyplot.savefig()` logic and see if the resulting plots scale well.

Reply:
`shap.plots.beeswarm` provide a `plot_size` argument, tuning the argument works, but it will also change the width/height ratio of the image. Like below. 
Haven't found other better solution yet.

![image](https://user-images.githubusercontent.com/19235986/145827281-480dbc60-35d5-4b9a-966b-ba3830399d40.png)

Reply:
Found another approach adjusting left marging by:
```
pyplot.subplots_adjust(left=0.4)
```

Then it works, but accurate adjusting require complex margin computation.
![image](https://user-images.githubusercontent.com/19235986/145829635-7cf0d514-b719-46b9-8388-327052b540f3.png)

Reply:
@WeichenXu123 Without adjustment, what do the plots look like?

Reply:
Like below, some text was truncated:
![image](https://user-images.githubusercontent.com/19235986/145913594-630667f1-5de4-4366-870e-103862af53b7.png)

Reply:
middle truncate name with max 20 chars .

Reply:
Looks great!